### DISCUSSION TITLE
How to use a service outside of react context application. eg a helper file

### DISCUSSION THREAD

**Discussion by anonymouscatcher** [statelyai/xstate#4071] [Q&A]
Hi,

I'm using XState in a React app. I'm utilizing createActorContext to share a service within the application. However, I have encountered an issue because I sometimes need to access the service outside of the React application, such as within nested machines or a helper file. Before using createActorContext, I had created a singleton class to initiate the service and obtain the service instance throughout the application, which was functioning properly. However, due to certain limitations, like using a web worker within one of the machines, I had to migrate to this new approach. This decision was made because I faced issues while building the app for edge runtime.
But long story short, I'm looking for a way to make this service globally available.

Thanks

**Comment by Andarist** [comment]
If you want to make your machine truly globally available then likely you shouldn't use `createActorContext`. Your previous approach was totally fine as well - you can always instantiate the actor outside of React and structure your application however you want. The only consideration for instantiating actors outside of React is that this approach isn't quite SSR-friendly. With SSR you need to somehow tie the actor's lifetime~ to a single request/render and React context is a good match for that.

**Reply by derek-primumco** [reply]
Thanks @Andarist! Just to clarify, you're suggesting `useMachine` in a root App file instead, and then passing around `state` and `send`?
